ipld/kamt/src/ext.rs

```diff
@@ -38,36 +38,36 @@ pub(crate) struct Extension {
 }
 
 impl Extension {
-    pub fn new(consumed: u8, path: Vec<u8>) -> Self {
-        Self { consumed, path }
+    pub fn new(length: u32, path: Vec<u8>) -> Self {
+        Self { length, path }
     }
 
-    pub fn consumed(&self) -> u8 {
-        self.consumed
+    pub fn len(&self) -> u32 {
+        self.length
     }
 
     pub fn path_bits(&self) -> HashBits {
-        HashBits::new_from_slice(&self.path)
+        HashBits::new_from_slice(&self.path, self.length)
     }
 
     pub fn is_empty(&self) -> bool {
-        self.consumed == 0
+        self.length == 0
     }
 
     /// See how many bits we can match of the path, consuming `bit_width` bits at a time.
     /// Return the total number of consumed bits, and actually consume them from the key.
-    pub fn longest_match(&self, hashed_key: &mut HashBits, bit_width: u32) -> Result<u8, Error> {
+    pub fn longest_match(&self, hashed_key: &mut HashBits, bit_width: u32) -> Result<u32, Error> {
         let mut path = self.path_bits();
-        let mut matched = 0u8;
-        while matched < self.consumed {
+        let mut matched = 0;
+        while matched < self.length {
             let consumed = hashed_key.consumed;
             let n1 = hashed_key.next(bit_width)?;
             let n2 = path.next(bit_width)?;
             if n1 != n2 {
                 hashed_key.consumed = consumed;
                 break;
             }
-            matched += bit_width as u8;
+            matched += bit_width;
```

Comment:
Here too, I never understood why `bit_width` is `u32` when we reject anything above 8.

Reply:
If we rejected everything above 256 exactly, I'd use a u8. Really, when I see someone taking a u8 as a value, I generally _assume_ that the entire range is likely valid because the programmer explicitly chose to use a u8 instead of the more normal u32.

Otherwise, the rule of thumb is to just use u32 for small numbers, u64 for large numbers, and usize for offsets.